code/ship/ship.cpp

```diff
@@ -10087,7 +10087,7 @@ int ship_create(matrix* orient, vec3d* pos, int ship_type, const char* ship_name
 		strcpy_s(shipp->ship_name, base_name);
 
 		// if generated name will be longer than allowable name, truncate the class section of the name by the overflow
-		int char_overflow = (strlen(base_name) + strlen(suffix)) - (NAME_LENGTH - 1);
+		int char_overflow = static_cast <int> (strlen(base_name) + strlen(suffix)) - (NAME_LENGTH - 1);
```

Comment:
I think it's a fine quick change to make. But static cast the individual strlen calls. m!m mentioned on a review in the past that putting more than one thing in a cast expression doesn't work as you would expect.

Reply:
Thanks for the heads up, that is excellent to know. I've updated so each is cast individually.